### ISSUE TITLE
How to add additional scrape config in GCP Managed Prometheus?

### ISSUE THREAD

**Issue by marandalucas** [k8spacket/k8spacket#18]
Hi folks, congratulation on this awesome project. 

I'm working on GKE with Managed Prometheus. I was able to deploy the k8spacket with helm and apparently, it's working properly.

```
2023/01/03 10:39:32 [nodegraph plugin] Connection: src=10.80.1.1 srcName=N/A srcPort=38732 srcNS= dst=10.80.1.71 dstName=pod.prometheus-6d4bdcb6f7-bdbgn dstPort=9090 dstNS=gmp-system closed=true bytesSent=111 bytesReceived=168 duration=1.9404e-05
2023/01/03 10:39:32 [nodegraph plugin] Connection: src=10.80.1.1 srcName=N/A srcPort=38730 srcNS= dst=10.80.1.71 dstName=pod.prometheus-6d4bdcb6f7-bdbgn dstPort=9090 dstNS=gmp-system closed=true bytesSent=109 bytesReceived=166 duration=0.000105884
2023/01/03 10:39:35 [nodegraph plugin] Connection: src=10.80.1.1 srcName=N/A srcPort=45910 srcNS= dst=10.80.1.145 dstName=pod.grafana-deployment-846c46f9f6-6p7hw dstPort=3000 dstNS=gmp-system closed=true bytesSent=113 bytesReceived=357 duration=1.5131e-05
2023/01/03 10:39:35 [nodegraph plugin] Connection: src=10.80.1.1 srcName=N/A srcPort=45908 srcNS= dst=10.80.1.145 dstName=pod.grafana-deployment-846c46f9f6-6p7hw dstPort=3000 dstNS=gmp-system closed=true bytesSent=113 bytesReceived=357 duration=0.000101168
```

Unfortunately, after the data sources have been configured properly, then the Dashboards don't show anything.

**I think I have to configure the additional scrape config in Managed Prometheus, but I am not able to do it.
Do you know how I can do it? Thanks in advance**

<a href="https://ibb.co/w4Wh3sL"><img src="https://i.ibb.co/L1tdb5P/Screenshot-2023-01-03-at-11-43-41.png" alt="Screenshot-2023-01-03-at-11-43-41" border="0"></a>
<a href="https://ibb.co/PjY6m7z"><img src="https://i.ibb.co/MShC2Y5/Screenshot-2023-01-03-at-11-43-50.png" alt="Screenshot-2023-01-03-at-11-43-50" border="0"></a>
<a href="https://ibb.co/QQStfrs"><img src="https://i.ibb.co/T2jdkmz/Screenshot-2023-01-03-at-11-44-06.png" alt="Screenshot-2023-01-03-at-11-44-06" border="0"></a>

**DASHBOARD**
<a href="https://ibb.co/jyp4X3D"><img src="https://i.ibb.co/fkc0wrH/Screenshot-2023-01-03-at-11-46-22.png" alt="Screenshot-2023-01-03-at-11-46-22" border="0"></a>

**Comment by k8spacket** [comment]
@marandalucas 
I assume the problem is connected to the wrong data source selected inside the Panel. Go to the Edit page:
<img width="227" alt="image" src="https://user-images.githubusercontent.com/95187481/210345761-ffb5ffb2-c9d7-450e-8fa4-d7df4502fac1.png">

and check the selected Data source:
<img width="672" alt="image" src="https://user-images.githubusercontent.com/95187481/210346568-650fca0e-e191-4d00-be91-6f1a76ec78fc.png">

Should be selected `node-graph-plugin` in your case (as it is in the explore screenshot)
The `Query string` should be filled by the value `namespace=$namespace&include=$include&exclude=$exclude&stats-type=$statstype`

**Comment by marandalucas** [comment]
Hi @k8spacket 

You are right! Now it's working. _The panel was using Prometheus Data Source (Default)._

This is my configuration. Apparently, the data sources should be configured automatically based on the JSON below:

`"type": "hamedkarbasi93-nodegraphapi-datasource"
`

**Is there something I missed? Thanks!**

```
---
apiVersion: integreatly.org/v1alpha1
kind: GrafanaDashboard
metadata:
  name: k8spacket-node-graph-dashboard
  namespace: gmp-system
  labels:
    app: grafana
spec:
  json:
    ""
  configMapRef:
    name: k8spacket-node-graph-dashboard
    key: k8spacket-node-graph-dashboard
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: k8spacket-node-graph-dashboard
  namespace: gmp-system
  labels:
    grafana_dashboard: "1"
    app.kubernetes.io/name: grafana
    dashboard-provider: default
data:
  k8spacket-node-graph-dashboard: |-
    {
      "annotations": {
        "list": [
          {
            "builtIn": 1,
            "datasource": "-- Grafana --",
            "enable": true,
            "hide": true,
            "iconColor": "rgba(0, 211, 255, 1)",
            "name": "Annotations & Alerts",
            "target": {
              "limit": 100,
              "matchAny": false,
              "tags": [],
              "type": "dashboard"
            },
            "type": "dashboard"
          }
        ]
      },
      "editable": true,
      "fiscalYearStartMonth": 0,
      "graphTooltip": 0,
      "id": 3,
      "iteration": 1654610569825,
      "links": [],
      "liveNow": false,
      "panels": [
        {
          "datasource": {
            "type": "hamedkarbasi93-nodegraphapi-datasource",
            "uid": "PC93AC6A0E760905D"
          },
          "gridPos": {
            "h": 31,
            "w": 24,
            "x": 0,
            "y": 0
          },
          "id": 2,
          "targets": [
            {
              "datasource": {
                "type": "hamedkarbasi93-nodegraphapi-datasource",
                "uid": "PC93AC6A0E760905D"
              },
              "queryText": "namespace=$namespace&include=$include&exclude=$exclude&stats-type=$statstype",
              "refId": "A"
            }
          ],
          "title": "Panel Title",
          "type": "nodeGraph"
        }
      ],
      "refresh": "",
      "schemaVersion": 35,
      "style": "dark",
      "tags": [],
      "templating": {
        "list": [
          {
            "current": {
              "selected": true,
              "text": "connection",
              "value": "connection"
            },
            "hide": 0,
            "includeAll": false,
            "label": "graph mode",
            "multi": false,
            "name": "statstype",
            "options": [
              {
                "selected": true,
                "text": "connection",
                "value": "connection"
              },
              {
                "selected": false,
                "text": "bytes",
                "value": "bytes"
              },
              {
                "selected": false,
                "text": "duration",
                "value": "duration"
              }
            ],
            "query": "connection,bytes,duration",
            "queryValue": "",
            "skipUrlSync": false,
            "type": "custom"
          },
          {
            "current": {
              "selected": false,
              "text": "",
              "value": ""
            },
            "description": "",
            "hide": 0,
            "label": "namespace (regexp)",
            "name": "namespace",
            "options": [
              {
                "selected": true,
                "text": "",
                "value": ""
              }
            ],
            "query": "",
            "skipUrlSync": false,
            "type": "textbox"
          },
          {
            "current": {
              "selected": false,
              "text": "",
              "value": ""
            },
            "hide": 0,
            "label": "include name (regexp)",
            "name": "include",
            "options": [
              {
                "selected": true,
                "text": "",
                "value": ""
              }
            ],
            "query": "",
            "skipUrlSync": false,
            "type": "textbox"
          },
          {
            "current": {
              "selected": false,
              "text": "",
              "value": ""
            },
            "hide": 0,
            "label": "exclude name (regexp)",
            "name": "exclude",
            "options": [
              {
                "selected": true,
                "text": "",
                "value": ""
              }
            ],
            "query": "",
            "skipUrlSync": false,
            "type": "textbox"
          }
        ]
      },
      "time": {
        "from": "now-6h",
        "to": "now"
      },
      "timepicker": {},
      "timezone": "",
      "title": "k8spacket - node graph",
      "weekStart": ""
    }
```

**Comment by marandalucas** [comment]
I had to update the json the following values

**Comment by k8spacket** [comment]
Hi @marandalucas 
Thanks for the information. I'll check if this uid is even necessary for dashboard definition files. Information about the type may be good enough.

**Comment by k8spacket** [comment]
Hi @marandalucas 
Fixed in PR #21 
Thanks.
